### PR TITLE
Refactored parts of the backup script to

### PR DESCRIPTION
**Pull request by ocraviotto**
+ add supported targets (besides original sqlite, now
  also files and directories)
+ improve variable semantics
+ improve overall readability
+ updated help text to hopefully make it easier to use

### CODE REVIEW

File: files/backup.sh [hunk 1]
```diff
@@ -116,20 +116,40 @@ process_opts()
...
+    [ "$#" -eq 2 ] || error "check_local_resource expects two arguments, $# given"
+    case "$2" in
+        sqlite|file)
+            [ -f "$1" ] || error "The file $1 does not exist as file. Check and try again"
+            ;;
+        directory)
+            [ -d "$1" ] || error "The directory $1 does not exist as directory. Check and try again"
+            ;;
+        *)
+            erro " Unrecognized source type. Must be one of \"sqlite\", \"file\" or \"directory\". Was: $2 "
```
Comment (stade) [inline files/backup.sh]: I guess this is a typo? I guess it should `error` instead of `erro`.

File: files/backup.sh [hunk 2]
```diff
@@ -142,26 +162,42 @@ dump_dir()
...
 
-    # Zip folder while keeping symlinks etc.
-    cd "$folder_path" && zip -y -r "$local_dump_file" .
+            cd "$local_source_path" && tar -czvf  "$local_dump_file" ./
 
-    log "...completed: dumped $folder_path to $local_dump_file"
+            log "...completed: archived and compressed $local_source_path into $local_dump_file"
+            ;;
+        *)
+            erro " Unrecognized source type. Must be one of \"sqlite\", \"file\" or \"directory\". Was: $backup_type "
```
Comment (stade) [inline files/backup.sh]: Same here `error` instead of `erro`?

File: files/backup.sh [hunk 3]
```diff
@@ -116,20 +116,40 @@ process_opts()
...
     fi
     s3cmd ls s3://"${s3_bucket}"/ &> /dev/null || error "Bucket check failed for s3://${s3_bucket}"
 }
 
-check_local_file()
+set_local_dump_file()
 {
-    [ -f "$1" ] || error "Checked for file $1 but it seems that it does not exist as such."
+    local tfile_prefix tfile_extension tfile_name
+    "$local_source_path" "$backup_type"
```
Comment (stade) [inline files/backup.sh]: I'm getting error from this line when running it in container in the test VM can you check this line I don't think it's correct. Maybe something is missing from: `"$local_source_path" "$backup_type"`

`/usr/local/bin/backup.sh: line 128: /var/lib/drone/drone.sqlite: Permission denied
`

File: files/backup.sh (hunk 2 above)
Comment (ocraviotto) [reply files/backup.sh]: Yes, typo

File: files/backup.sh (hunk 3 above)
Comment (ocraviotto) [reply files/backup.sh]: This is a left over from a comment, they do not go there... deleting

**Pull request merged**